Library/Homebrew/dev-cmd/contributions.rb

```diff
@@ -169,8 +171,16 @@ def scan_repositories(repos, person, args)
 
       puts "Determining contributions for #{person} on #{repo_full_name}..." if args.verbose?
 
+      commits_authored = GitHub.repo_commit_count_for_user(repo_full_name, person, "author", args)
+      commits_committed = GitHub.repo_commit_count_for_user(repo_full_name, person, "committer", args)
+      # Only count committers where the author is not the same person. Avoid negative numbers or subtracting zero.
+      if commits_authored.positive? && commits_committed.positive?
+        commits_committed = commits_authored - commits_committed
+      end
```

Comment:
Am I right in understanding that:
| Author       | Committer    | Counted In |
|--------------|--------------|------------|
| `user`         | other or `user` | `commits_authored`     |
| other or `user` | `user`        | `commits_committed`  |

are all the possibilities? So `commits_committed = commits_authored - commits_committed` doesn't seem correct, right? For example, let's say I have exactly 5 commits in homebrew-core:
* Author – `nandahkrishna`; Committer – `BrewTestBot`
* Author – `nandahkrishna`; Committer – `nandahkrishna`
* Author – `nandahkrishna`; Committer – `nandahkrishna`
* Author – `samford`; Committer – `nandahkrishna`
* Author – `issyl0`; Committer – `nandahkrishna`

We would finally have `commits_authored = 3` and `commits_committed = -1` (originally `4`, but `commits_committed = 3 - 4 = -1`). There aren't too many commits of the last kind especially after auto-merge was introduced though.

Reply:
Maybe my 10pm maths was flawed too. I did take my high school maths exam three times. 🙈 I'll work all this out again.

Reply:
What I was trying to get to was...

Assume I have 4 commits in Cask:

| Author | Committer |
| issyl0 | BrewTestBot |
| issyl0 | reitermarkus |
| issyl0 | commitay |
| issyl0 | BrewTestBot |

`commits_authored = 4`, `commits_committed = 0`, so `commits_committed` was coming out as 4 (4 - 0 = 4) before the `.positive?` check (`> 0`).

I can see now how this would be flawed for your numbers. Maybe this needs some tests with a variety of different data. 😅

I guess we could do this the "proper" way by getting the SHAs of the authored commits and the SHAs of the committed commits and getting the uniques in each and counting _those_ to deduplicate. But that seemed like a lot.

Reply:
[693d609](https://github.com/Homebrew/brew/pull/14860/commits/693d609730e5f20c0347000850583506c825abae) should fix this, but please check my maths again!